### ISSUE TITLE
Repo cleanup post-changes

### ISSUE THREAD

**Issue by manulera** [pombase/fypo#4328]
Hello @kimrutherford and @ValWood.

I have made a bunch of changes in the repository, following some help from the obo people.

I think we no longer need to run the release procedure locally, and we can do it using github actions.

There are two separate actions:

* [refresh imports](https://github.com/pombase/fypo/actions/workflows/update-imports.yml):
    * Updates the imports into a single file [merged_import](https://github.com/pombase/fypo/blob/master/src/ontology/imports/merged_import.owl) instead of using all the separate files that was using below.
    * Creates a pull request that we then have to accept to merge into master.
    * This should be done before the other action, or when new terms from an ontology, after commiting the new terms to the `.txt` files (e.g. chebi_terms.txt). It takes ~40 mins to run.
* [run release](https://github.com/pombase/fypo/actions/workflows/make_release.yml):
    * Runs the release pipeline and creates a draft release with all the release artifacts uploaded. See this [draft release](https://github.com/pombase/fypo/releases/tag/untagged-7fb7411529b81f557a54) that I created yesterday.

The release action does not commit anything to the repository, and that is because other ontologies do not keep those release artifacts in source control, only the equivalent to fypo-edit. We have already moved fypo.owl and others out of source control because they became too big, so I think it would be safe to remove all the other release artifacts as well, and keep them only as release attachments. However, we have to think if us or someone else is using these files from the repo url and not the release url. We already ammended some urls when we moved fypo.owl out, is there anything else that comes to mind?

In addition, I will take the occasion to remove some files that are no longer needed, e.g. all the `.owl` and `.obo` files from the imports folder except for `merged_import.owl`. 

cc: @matentzn does this make sense? Anything else we could clean up since we are doing it?

**Comment by manulera** [comment]
@matentzn what about the files in this folder https://github.com/pombase/fypo/tree/master/imports should a `merged_import.owl` be committed there as well? These were updated with the former release pipeline, but the update imports from now seems to only affect this file: https://github.com/pombase/fypo/blob/master/src/ontology/imports/merged_import.owl

Then, we used to get reports in `reports/fypo-edit.owl-obo-report.tsv` and `src/ontology/reports/fypo-edit.owl-obo-report.tsv` (identical). What should we do about those?

**Comment by ValWood** [comment]
Brillaint, thanks for doing this, and thanks @matentzn and  @gouttegd for the help!

**Comment by kimrutherford** [comment]
Great work!  That's a big improvement.

**Comment by manulera** [comment]
Hi @kimrutherford @matentzn.

* Kim: Do we use urls of ontologies in the repo that would be moved to the release attachments?
* Nico: Any thoughts on who should be warned about the changes in the structure of the repo? Someone that does not uses purl obolibrary urls?

**Comment by matentzn** [comment]
@manulera I think it is sufficient to add a prominent warning on the release notes on GitHub.

Remind people to.

Never
ever
ever
....

use URLs to refer to data and ontology files on the web.

Always use PURLs and only PURLs!

**Comment by kimrutherford** [comment]
> We have already moved fypo.owl and others out of source control because they became too big, so I think it would be safe to remove all the other release artifacts as well, and keep them only as release attachments

Hi @manulera

Could you explain what you mean by "release attachments".  I don't understand.

> Kim: Do we use urls of ontologies in the repo that would be moved to the release attachments?

We use these OBO files, which we read directory from a local checkout of the fypo repository:

  - supplemental_files/fypo_extension_relations.obo
  - fypo-simple-pombase.obo
  - fyeco.obo

> Never ever ever ....
> use URLs to refer to data and ontology files on the web.

We definitely don't do that.  :100:

> Always use PURLs and only PURLs!

Oh dear, we don't do that either.  :frowning_face:

**Comment by matentzn** [comment]
Local checkout is also not reliable, because file locations change within a repo.. if you use purls and curl the ontology from the PURL, than you can be certain that you will always grab the correct file

**Comment by matentzn** [comment]
The release and imports date are very different things! Dont worry, this is exactly as it should be. For the import, the date denotes when the import was last refreshed.

**Comment by ValWood** [comment]
Hi,
 I tried to run the new release pipeline for the first time today.

I got an error on refresh imports
https://github.com/pombase/fypo/actions/runs/5201493460/jobs/9381726725#step:7:237

@matentzn if you have any info about the error codes please let me know,  (@manulera is on vacation so I am stuck until
he returns).

I also posted a question on Slack. 

Thanks if you can advise.

**Comment by ValWood** [comment]
From Nico:
 the problem is that you have not enough memory set tun run the merged import, see https://oboacademy.github.io/obook/howto/odk-setup/#problems-with-memory-important

**Comment by matentzn** [comment]
Sorry I should have read this better. You did not run on your machine, you ran in GH actions.. See slack, this is a much bigger problem, and I am not sure if it can be solved at all.. is @manulera not there?

**Comment by ValWood** [comment]
He is in Japan...

**Comment by ValWood** [comment]
Manu has run it successfully though....

**Comment by matentzn** [comment]
Yea he may have been lucky though. He must have been close to the memory limit..

**Comment by kimrutherford** [comment]
I've committed a change which seems to have helped: https://github.com/pombase/fypo/commit/61e36f45f43dc9c919ded475751518c6117eb7e7

The "Refresh imports" Action finished after that: https://github.com/pombase/fypo/actions/runs/5204542287

**Comment by manulera** [comment]
Bravooo @kimrutherford

**Comment by ValWood** [comment]
excellent,  thanks Kim

**Comment by ValWood** [comment]
The second part of the pipeline ran OK.
